tests/packages/sharedlib-in-package/mypkg/__init__.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-FileCopyrightText: 2024 The meson-python developers
+#
+# SPDX-License-Identifier: MIT
+
+import os
+import sys
+
+
+# Note: the below function is used in the docs via a literalinclude directive.
+#       When making any changes to this file, please ensure the right set of
+#       lines is pulled into the docs, and adjust the :lines: attribute of
+#       the literalinclude if needed.
+
+
+def _enable_sharedlib_loading():
+    """
+    Ensure the shared libraries in this dir and the `sub` subdir can be loaded
+    on Windows.
+
+    One shared library is installed alongside this ``__init__.py`` file.
+    Windows can load it because it searches for DLLs in the directory a ``.pyd``
+    (Python extension module) is located in. Cygwin does not though. For a
+    shared library in another directory inside the package, Windows also needs
+    a hint.
+
+    This function is Windows-specific due to lack of RPATH support on Windows.
+    It cannot find shared libraries installed within wheels unless we either
+    amend the DLL search path or pre-load the DLL.
+
+    Note that `delvewheel` inserts a similar snippet into the main
+    `__init__.py` of a package when it vendors external shared libraries.
```

Comment:
Is this really relevant for users? The topic is already complex and I would like to avoid to overwhelm the readers with information.

Reply:
I'd keep it - there is no other documentation that explains any of this well, so I'd err on the side of completeness. Users reading this are probably fairly technical (but this is still new info to many of them), so I'm not too worried about explaining too much.